lib/archethic/transaction_chain.ex

```diff
@@ -793,16 +793,39 @@ defmodule Archethic.TransactionChain do
   end
 
   defp do_fetch_transaction_chain(nodes, address, paging_state, opts \\ []) do
+    order = Keyword.get(opts, :order, :asc)
+
     conflict_resolver = fn results ->
       results
       |> Enum.sort(
-        &((&1.more? and !&2.more?) or length(&1.transactions) > length(&2.transactions))
+        # Prioritize more? at true
+        # then length of transaction list
+        # then regarding order, the oldest or newest transaction timestamp
+        # of the first element of the list
+        &with false <- &1.more? and !&2.more?,
+              false <- length(&1.transactions) > length(&2.transactions) do
+          if Enum.empty?(&1.transactions) do
+            false
+          else
+            case order do
```

Comment:
We could factorize to reduce boilerplate. 
And use the case to determine the right comparison operator.